content/code_comments/introduction-to-code-commenting/_index.md

```diff
@@ -0,0 +1,247 @@
+---
+content_type: topic
+ready: true
+title: Understanding Effective Comments in Coding
+---
+
+## Introduction
+
+In programming, comments play a crucial role in code documentation. They provide explanations, clarifications, and additional context to make the code more understandable for developers. However, not all comments are created equal. It is important to distinguish between good comments and bad comments to ensure effective code communication. Let's explore what makes a comment good or bad.
+
+## Characteristics of a Good Comment
+
+- Purposeful Explanation: A good comment serves the purpose of explaining complex logic, algorithms, or business rules. It provides insights into the code's functionality and helps other developers understand the intentions behind certain decisions.
+
+  ```python
+  # Function to calculate the factorial of a number using recursion
```

Comment:
There are a few things that are bad about this comment:

1. It should be a docstring so that Python's help system picks it up. 
2. Overly fancy - generally speaking it's best to explain things in  the simplest way possible. The goal is to be understood. The goal is not to try to look smart.  

I actually see learners trying to look clever by using overly stiff language and big words a lot. This might be worth sharing with them: https://socialtriggers.com/the-big-problem-with-big-words-hint-they-make-you-look-stupid/